fastjet/internal/delaunay/point.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/gonum/floats"
+)
+
+// Point in the X-Y Plane, that holds dynamic information about the
+// adjacent triangles, the nearest neighbor and the distance to that neighbor
+type Point struct {
+	// X and Y are the coordinates of the point.
+	X, Y float64
+	// adjacentTriangles is used for the triangulation, to find the nearest neighbor and
+	// to find the Voronoi Cell.
+	adjacentTriangles triangles
+	// isOutside indicates whether the point is outside the triangulation.
+	isOutside bool
+	// ID is a user defined identifier. There are no restrictions on how the user uses it.
+	// The field is used when points are removed. Copies of the points around the point to
+	// be removed are made. The ID is set incremental in counterclockwise order. It identifies
+	// the original. It is also used to determine if a Triangle is inside or outside the
+	// polygon formed by all those points.
+	ID      int
+	nearest *Point
+	// squared distance to the nearest neighbor
+	dist float64
+}
+
+// NewPoint returns Point for the given x,y coordinates and id
+func NewPoint(x float64, y float64, id int) *Point {
```

Comment:
@vladimir-ch Can I just make the ID a private field and increment it for each point inserted? I'm not quite sure how you were thinking of implementing this with the interface variable p.